src/test_parser.rs

```diff
@@ -181,10 +181,13 @@ fn parse_list_errors() -> Result<(), Box<dyn Error>> {
 
 #[test]
 fn parse_inner_list_errors() -> Result<(), Box<dyn Error>> {
-    let input = "c b); a=1";
```

Comment:
> https://www.rfc-editor.org/rfc/rfc9651.html#section-4.2.1.2
> Consume the first character of input_string; if it is not "(", fail parsing.

I'd prefer to keep these unit tests. They check that functions behave correctly, and that we implement the RFC line by line.

Reply:
If you feel strongly, I can continue using `pub(crate)` on these methods and add `#[should_panic]` tests that run under debug mode, but I find that testing these private functions makes the code brittle and difficult to refactor, as evidenced by some of the changes I've had to make in my other PRs.

Reply:
> Any specific reason these methods wouldn't be pub(crate) ?

They aren't called by anything outside this file.

Reply:
I'd still rather keep this test 🙂